libs/components/src/next/form/radio/vl-radio-group.component.ts

```diff
@@ -0,0 +1,224 @@
+            this.checkRadioForValue(this.value);
+        }
+
+        if (changedProperties.has('name')) {
+            this.updateRadiosForAttribute('name');
+        }
+
+        if (changedProperties.has('block')) {
+            this.updateRadiosForAttribute('block');
+        }
+
+        if (changedProperties.has('readonly')) {
+            this.updateRadiosForAttribute('readonly');
+            this.preventDefaultEventOnAllInputsOfAllRadio(this.readonly);
+        }
+
+        if (changedProperties.has('disabled')) {
+            this.updateRadiosForAttribute('disabled');
+        }
+
+        if (changedProperties.has('error')) {
+            this.updateRadiosForAttribute('error');
+        }
+
+        if (changedProperties.has('isInvalid')) {
+            this.getRadios()?.forEach((radio) => this.setAttributeForCondition(radio, 'error', this.isInvalid));
+        }
+
+        if (changedProperties.has('success')) {
+            this.updateRadiosForAttribute('success');
+        }
+    }
+
+    render(): TemplateResult {
+        return html` <slot></slot> `;
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+
+        this.removeEventListener('vl-checked', this.updateGroupAfterCheck);
+    }
+
+    resetFormControl(): void {
+        super.resetFormControl();
+
+        this.resetAllRadios();
+
+        this.resetToInitialValue();
+    }
+
+    get validationTarget(): HTMLInputElement | undefined | null {
+        const firstRadio = this.getRadios()[0];
+        return firstRadio ? firstRadio.validationTarget : null;
+    }
+
+    private resetToInitialValue(): void {
+        this.setValue(this.initialValue);
+        this.checkRadioForValue(this.initialValue);
+        this.value = this.initialValue;
+    }
+
+    private checkRadioForValue(value: string | null): void {
+        this.getRadios()?.forEach((radio) => {
+            if (value && radio.getAttribute('value') === value) {
+                radio.setAttribute('checked', '');
+            } else {
+                radio.removeAttribute('checked');
+            }
+        });
+    }
+
+    private resetAllRadios(): void {
+        this.getRadios()?.forEach((radio) => {
+            radio.removeAttribute('checked');
+        });
+    }
+
+    private preventDefaultAndFocus = (event: Event) => {
+        const targetElement = event.target as VlRadioComponent;
+        targetElement?.validationTarget?.focus();
+        event.preventDefault();
+    };
+
+    private preventDefaultEventOnAllInputsOfAllRadio(readonly: boolean): void {
+        this.getRadios()?.forEach((radio) => {
+            if (readonly) {
+                radio.addEventListener('click', this.preventDefaultAndFocus);
+            } else {
+                radio.removeEventListener('click', this.preventDefaultAndFocus);
+            }
+        });
+    }
+
+    private setAttributeForCondition = (radio: VlRadioComponent, attribute: string, condition: boolean) => {
```

Comment:
Ik zou deze methode verwijderen, je gebruikt die maar 2 keer en het zorgt voor extra complexie. Dit voluit schrijven op de 2 plaatsen waar je het gebruikt is leesbaarder.